sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedContextClientImpl.java

```diff
@@ -124,11 +124,25 @@ public <T> Flux<FeedResponse<T>> createDocumentChangeFeedQuery(CosmosAsyncContai
                                                                    CosmosChangeFeedRequestOptions changeFeedRequestOptions,
                                                                    Class<T> klass) {
 
+        return this.createDocumentChangeFeedQuery(collectionLink, changeFeedRequestOptions, klass, true);
+    }
+
+    @Override
+    public  <T> Flux<FeedResponse<T>> createDocumentChangeFeedQuery(CosmosAsyncContainer collectionLink,
+                                                                    CosmosChangeFeedRequestOptions changeFeedRequestOptions,
+                                                                    Class<T> klass,
+                                                                    boolean isSplitHandlingDisabled) {
+
+        // Case 1: when split handling should be disabled
         // ChangeFeed processor relies on getting GoneException signals
         // to handle split of leases - so we need to suppress the split-proofing
         // in the underlying fetcher/pipeline for the change feed processor.
+        // Case 2: when split handling should be enabled
+        // A ChangeFeedProcessor instance which is backed by a client with a stale
+        // PKRange cache will run into 410/1002s (PartitionKeyRangeGone) if disable split handling is true
+        // in getCurrentState and getEstimatedLag scenarios therefore disable split handling should explicitly be set to false
         CosmosChangeFeedRequestOptions effectiveRequestOptions =
-            ModelBridgeInternal.disableSplitHandling(changeFeedRequestOptions);
+            isSplitHandlingDisabled ? ModelBridgeInternal.disableSplitHandling(changeFeedRequestOptions) : changeFeedRequestOptions;
```

Comment:
should we just always do `ModelBridgeInternal.disableSplitHandling(isSplitHandlingDisabled)`

Reply:
Fixed in next iteration.